### DISCUSSION TITLE
Microsoft BUILD 2022 Questions/Feedback

### DISCUSSION THREAD

**Discussion by paulgildea** [microsoft/fluentui#23183] [General]
If you have questions or feedback for Fluent UI React associated with Microsoft BUILD feel free to add them here!

Thank you and have a great BUILD!

**Comment by SIGAN** [comment]
Why so many components from v8 are not migrated to v9?
Things like Nav, Dialog, Calendar, DatePicker are the core building blocks. And none is available.
According to the release schedule, there is no time left for anything but bug fixes.

That table for the component mapping basically is a nightmare for any frontend dev, because so many components are missing:
https://react.fluentui.dev/?path=/docs/concepts-upgrading-from-v8-component-mapping--page

**Reply by khmakoto** [reply]
Hi @SIGAN, v9 is a complete revamp of the library that has fundamental and structural changes from v8 for performance and optimization reasons. Given that, it was always an impossible task to ship a next version that had parity with v8 in terms of the number of components it exports. What we are shipping with v9 is a set of "baseline" components. That is, the primitives, most used and building blocks for more complex components. This allows us to have a reasonable timeframe and product-test the library.

The good news is that we have created v9 to be able to be used side-by-side with v8 controls (albeit with some things of note), so, for anything not existing in v9 yet, you can feel free to use your existing v8 solution. Likewise, for everything you already have in v8, you can pick and choose your path for an incremental upgrade.

In the future, we'll be looking at implementing more and more components to (eventually) achieve parity with v8. How long this will take I can't tell you, that's a question that has a lot of variables including resourcing and business need. We are, however, always receptive of customer needs and our decisions of which components to implement with a higher priority depend a lot on first and third party feedback, so please feel free to tell us what you need the most.

I hope this answer eases your concerns, and I'm always happy to talk more/help you in the migration/upgrade/intake of v9 in your product.

**Comment by raymeskhoury** [comment]
Just checking out the new components and I'm interested in testing the Toolbar as a replacement for CommandBar. However it seems that the ToolbarButton component isn't properly packaged? The codesandbox links aren't working off here: https://react.fluentui.dev/?path=/docs/preview-components-toolbar--default

See e.g. https://codesandbox.io/s/zutwj2?file=/example.tsx 

Also it may be worth noting in the [v8->v9 component mapping](https://react.fluentui.dev/?path=/docs/concepts-upgrading-from-v8-component-mapping--page) that Toolbar is the successor to CommandBar (at least that's what I'm assuming).

**Reply by paulgildea** [reply]
Thanks for reaching out @raymeskhoury. Would you mind filing an issue for the packaging problem with `Toolbar`. 

@GeoffCox can we add an entry in the component mapping for CommandBar to Toolbar?

**Reply by raymeskhoury** [reply]
Done in #24015

**Comment by yugi0h** [comment]
Does `table component` future can resize or fit size column ?

**Comment by sgbdev** [comment]
Just wanted to say thanks for all the thought and work that went into v9! I love using Griffel, and using theme tokens have made leveraging the theme feel a lot more intuitive and straightforward.

**Comment by marwan38** [comment]
We have been using `react-northstar` internally and we have been following the standards set within the source code of northstar. This makes our migration to v9 cumbersome. It won't happen till a few months from now, at least. We maintain a similar folder structure for components and theming. V9 changes that drastically. Things like `useTelemtry`, `useStyles`, slots, shorthand api, etc.. all done exactly as it is by v0's internals.

1. Why is northstar staying as it's own thing?
2. Will teams be migrating to v9?
3. Is React18 support on the roadmap? They have a new hook for css-in-js and there's the world of streaming that 18 opens us to.
4. Are there any reasons not to migrate from v0 to v9?
5. Does v0 plan to stay with it's current styling engine (fela), or are there plans to change?

I would/do have a ton more questions, but, this is what I have from the top of my head at the moment.

**Reply by jurokapsiar** [reply]
Hi,

It is recommended to migrate to v9 if you are using v0. Please see https://react.fluentui.dev/?path=/docs/concepts-migration-overview--page

v9 implements updated designs and significantly better performance. Teams is also migrating to v9. There shouldn't be any reason not to migrate to v9. V9 does not yet have full set of components, so the migration would need to be gradual.

There are currently no plans to implement new features in react-northstar. But external contributions are welcome if they would help your scenarios.

**Comment by umarshahh** [comment]
Hi Team, 
Im planning to use some preview component in one of my projects, are they unstable or will I be able to use them normally like other components? Will they create issues in my project? [Components like Data table, Combo boxes, dropdowns and alerts]

**Reply by micahgodbolt** [reply]
Unstable controls are mostly feature complete, but they are not API stable. You are welcome to start using them, but be aware that APIs and features may change before final release

**Reply by paulgildea** [reply]
Here's a high level overview of component lifecycle: https://dev.to/paulgildea/fluent-ui-react-component-lifecycle-29n5

**Comment by HolyChen** [comment]
Hello Team,
In my project, `<Stack>` and `<StackItem>` are used widely, they are convenient to create new components and construct new pages for us. And because it is a library component, we encourage our team to use `<Stack>` rather than `<div>` as far as possible. Now FluentUI v9 removed `<Stack>` and encourage developers to manage layout using `<div>`, I'm curious of why FluentUI team made this decision, and what benefits developers will get from relacing `<Stack>` with `<div>`. Thank you.

**Reply by savkelita** [reply]
I totally agree with you, we also have the same problem. Is there any alternative way (component, library) or suggestion?

**Comment by Acyapinsa** [comment]
Hi are there any examples of how to change the theme dynamically like v8? Thanks

**Comment by siilianttivermas** [comment]
Hi, I'm trying to make a similar component which was NormalPeoplePicker in version 8. The thing that is not possible to do with ComboBox is to have those tag's of selected items inside the input element. This image from version 9 site suggest that it would be possible but it is not. Is the functionality coming perhaps in the future? This kind of people picking is really used feature in our software.
<img width="804" alt="image" src="https://github.com/microsoft/fluentui/assets/131360437/bc719a7b-b8ab-41d3-ba40-7a5ef8eae4e8">

**Comment by rickreilly** [comment]
Is it possible to run Fluent V9 on React 16?
